src/NotePad/MenuStrategy.java

```diff
@@ -0,0 +1,60 @@
+import javax.swing.*;
+import java.nio.file.Paths;
+
+public class MenuStrategy {
+    Recent r;
+    MenuStrategy(JMenu recent){
+        Recent r1 = new Recent(recent);
+        r=r1;
+    }
+     public void menuCall(String type, JTextPane d,UiStructure ui,JMenuItem adr1) {
+        /*
+        * Through strategy, call different functions while different action was requested
+        * */
+         switch (type) {
```

Comment:
might want to decouple the strategy instantiation from this class, and avoid switch statements if possible